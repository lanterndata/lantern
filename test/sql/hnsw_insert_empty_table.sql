CREATE TABLE small_world (
    id varchar(3),
    vector real[]
);

CREATE INDEX ON small_world USING hnsw (vector);

SELECT * FROM small_world WHERE 1=1;

INSERT INTO small_world (id, vector) VALUES ('xxx', '{0,0,0}');

-- make sure NULL inserts into the index are handled correctly
INSERT INTO small_world (id, vector) VALUES ('xxx', NULL);
\set ON_ERROR_STOP off
INSERT INTO small_world (id, vector) VALUES ('xxx', '{1,1,1,1}');
\set ON_ERROR_STOP on
