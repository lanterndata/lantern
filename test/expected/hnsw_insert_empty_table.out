CREATE EXTENSION IF NOT EXISTS lanterndb;
NOTICE:  extension "lanterndb" already exists, skipping
DROP TABLE IF EXISTS small_world;
CREATE TABLE small_world (
    id varchar(3),
    vector real[]
);
CREATE INDEX ON small_world USING hnsw (vector);
INFO:  done init usearch index
INFO:  inserted 0 elements
INFO:  done saving 0 vectors
SELECT * FROM small_world WHERE 1=1;
 id | vector 
----+--------
(0 rows)

INSERT INTO small_world (id, vector) VALUES ('xxx', '{0,0,0}');
-- make sure NULL inserts into the index are handled correctly
INSERT INTO small_world (id, vector) VALUES ('xxx', NULL);
\set ON_ERROR_STOP off
INSERT INTO small_world (id, vector) VALUES ('xxx', '{1,1,1,1}');
ERROR:  Wrong number of dimensions: 4 instead of 3 expected
\set ON_ERROR_STOP on
